From 27aa646db0f0465c5abf8e5cd545e070e7f14120 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 26 May 2021 15:35:26 +0900
Subject: [PATCH] cifsd: fix wrong return value in ksmbd_crypt_message()
Git-commit: 27aa646db0f0465c5abf8e5cd545e070e7f14120
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Change error return instead of returning always success return.

Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/auth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifsd/auth.c b/fs/cifsd/auth.c
index 711f8dec38e1..5a56dd65fa06 100644
--- a/fs/cifsd/auth.c
+++ b/fs/cifsd/auth.c
@@ -1253,7 +1253,7 @@ int ksmbd_crypt_message(struct ksmbd_conn *conn, struct kvec *iov,
 				      key);
 	if (rc) {
 		ksmbd_err("Could not get %scryption key\n", enc ? "en" : "de");
-		return 0;
+		return rc;
 	}
 
 	if (conn->cipher_type == SMB2_ENCRYPTION_AES128_GCM ||
-- 
2.33.0

